nixos/modules/services/web-apps/mealie.nix

```diff
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ...}:
+let
+  cfg = config.services.mealie;
+  pkg = pkgs.mealie;
+
+  # TODO Fix ingredient quantities NLP recognition
+in
+{
+  options.services.mealie = {
+    enable = lib.mkEnableOption "Mealie, a recipe manager and meal planner";
+
+    listenAddress = lib.mkOption {
+      type = lib.types.str;
+      default = "0.0.0.0";
+      description = "Address on which the service should listen.";
+    };
+
+    port = lib.mkOption {
+      type = lib.types.port;
+      default = 9000;
+      description = "Port on which to serve the Mealie service.";
+    };
+
+    settings = lib.mkOption {
+      type = lib.types.attrs;
```

Comment:
This isn't critical but note that RFC42 includes a more specific type for freeform options:

https://github.com/NixOS/rfcs/blob/master/rfcs/0042-config-option.md#part-1-1